mesa.rb

```diff
@@ -5,40 +5,36 @@ class Mesa < Formula
   sha256 "bab24fb79f78c876073527f515ed871fc9c81d816f66c8a0b051d8d653896389"
 
   option "without-test", "Skip compile-time tests"
-  option "with-static",   "Build static libraries (not recommended)"
+  option "with-static", "Build static libraries (not recommended)"
 
-  # DEPENDENCIES
   depends_on "pkg-config" => :build
-
-  # Python and it Mako module
   depends_on :python => :build
-  resource "mako" do
-    url "https://pypi.python.org/packages/7a/ae/925434246ee90b42e8ef57d3b30a0ab7caf9a2de3e449b876c56dcb48155/Mako-1.0.4.tar.gz"
-    sha256 "fed99dbe4d0ddb27a33ee4910d8708aca9ef1fe854e668387a9ab9a90cbf9059"
-  end
-
   depends_on "flex"  => :build
   depends_on "bison" => :build
-
-  depends_on "xorg" # TODO: figure out which packages exactly are used and depend on them only
+  depends_on "xorg"
   depends_on "libdrm"
-  depends_on "systemd"
+  depends_on "libsha1"
+  depends_on "systemd" # provides libudev <= needed by "gbm"
   depends_on "llvm"
-  depends_on "libelf" # ?
+  depends_on "libelf" # radeonsi requires libelf when using llvm
   depends_on "libomxil-bellagio"
-  depends_on "wayland" => :recommended # => HEAD is preferred
+  depends_on "wayland" => :recommended
+  depends_on "valgrind" => :recommended
+  depends_on "libglvnd" => :optional
+  depends_on "libva" => :recommended
+  depends_on "autoconf" => :build if build.with?("libva") && build.without?("wayland")
 
```

Comment:
```
$ brew unlink libvdpau; brew reinstall https://github.com/maxim-belkin/homebrew-xorg/raw/mesa-update/mesa.rb
Unlinking /data/linuxbrew/Cellar/libvdpau/1.1.1... 8 symlinks removed
######################################################################## 100.0%
==> Reinstalling mesa
==> Downloading ftp://ftp.freedesktop.org/pub/mesa/12.0.1/mesa-12.0.1.tar.xz
Already downloaded: /root/.cache/Homebrew/mesa-12.0.1.tar.xz
==> Downloading http://www.linuxfromscratch.org/patches/blfs/svn/mesa-12.0.1-add_xdemos-1.patch
Already downloaded: /root/.cache/Homebrew/mesa--patch-53492ca476e3df2de210f749983e17de4bec026a904db826acbcbd1ef83e71cd.patch
==> Patching
==> Applying mesa-12.0.1-add_xdemos-1.patch
patching file xdemos/glinfo_common.c
patching file xdemos/glinfo_common.h
patching file xdemos/glxgears.1
patching file xdemos/glxgears.c
patching file xdemos/glxinfo.1
patching file xdemos/glxinfo.c
patching file xdemos/Makefile
==> Downloading https://pypi.python.org/packages/7a/ae/925434246ee90b42e8ef57d3b30a0ab7caf9a2de3e449b876c56dcb48155/Mako-1.0.4.tar.gz
Already downloaded: /root/.cache/Homebrew/mesa--mako-1.0.4.tar.gz
==> python -c import setuptools... --no-user-cfg install --prefix=/data/linuxbrew/Cellar/mesa/12.0.1/libexec/vendor --single-version-externally-managed --record=installed.txt
==> ./autogen.sh CFLAGS=-O2 CXXFLAGS=-O2 --prefix=/data/linuxbrew/Cellar/mesa/12.0.1 --sysconfdir=/data/linuxbrew/etc --localstatedir=/data/linuxbrew/var --disable-dependency-tracking --disable-silent-rules --enable-dri --enable-dri3 --enable-gallium-tests --enable-galliu
Last 15 lines from /root/.cache/Homebrew/Logs/mesa/02.autogen.sh:
checking for RADEON... yes
checking for mincore... yes
checking for VL... yes
checking for XVMC... yes
checking for VDPAU... no
configure: error: Package requirements (vdpau >= 1.1) were not met:

No package 'vdpau' found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables VDPAU_CFLAGS
and VDPAU_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.
```